core/http-auth-event-stream/src/main/java/software/amazon/awssdk/http/auth/eventstream/internal/DefaultAwsV4EventStreamHttpSigner.java

```diff
@@ -15,27 +15,43 @@
 
 package software.amazon.awssdk.http.auth.eventstream.internal;
 
+import static software.amazon.awssdk.http.auth.internal.util.SignerConstant.X_AMZ_CONTENT_SHA256;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.CompletableFuture;
+import org.reactivestreams.Publisher;
 import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.http.SdkHttpRequest;
 import software.amazon.awssdk.http.auth.eventstream.AwsV4EventStreamHttpSigner;
-import software.amazon.awssdk.http.auth.spi.AsyncSignRequest;
-import software.amazon.awssdk.http.auth.spi.AsyncSignedRequest;
-import software.amazon.awssdk.http.auth.spi.SyncSignRequest;
-import software.amazon.awssdk.http.auth.spi.SyncSignedRequest;
-import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+import software.amazon.awssdk.http.auth.internal.DefaultAwsV4HttpSigner;
+import software.amazon.awssdk.http.auth.internal.checksums.ContentChecksum;
 
 /**
  * A default implementation of {@link AwsV4EventStreamHttpSigner}.
  */
 @SdkInternalApi
-public class DefaultAwsV4EventStreamHttpSigner implements AwsV4EventStreamHttpSigner {
+public final class DefaultAwsV4EventStreamHttpSigner extends DefaultAwsV4HttpSigner implements AwsV4EventStreamHttpSigner {
```

Comment:
> `extends DefaultAwsV4HttpSigner`

We've found that maintaining cross-module APIs with inheritance creates a lot harder-to-follow code than maintaining cross-module APIs with composition. People don't think very hard about the API contracts for inheritance as much as they think about it for APIs. And absolutely everything is an API contract in inherited types that cross module boundaries - no renaming of protected fields, no changing implicit or explicit behavior around protected methods.

Let's keep going with this approach, but I see a lot of internal coupling that might be tricky to support in the future, since this relationship is cross-module. Once the churn on this stuff has died down, let's see if there's a clean abstraction using composition instead of inheritance.

Reply:
I was going for that initially, so I agree. There was feedback about reducing code duplication across signers, which inheritance naturally solves when lots of commonality (but adds coupling obviously). I'd definitely prefer composition.

Reply:
I know we are discussing refactoring, but with this current approach for now, this means we have to change the `DefaultAwsV4HttpSigner` to `@SdkProtectedApi`.